### PR TITLE
Add achievement feature

### PR DESCRIPTION
**Pull request by ChubbsBunns**
## Add Achievement Feature
#58 
#113 

## Fitness Duke is able to now reward users with built-in achievements.
These achievements include hitting milestones of all the different workouts, each with their own name and difficulty level and requirements.

### List achievements
Users can now list all the existing achievements, exposing their difficulty level and whether they have completed or not

### Congratulations message
Upon completing an achievement when finishing a workout, users will be greeted with a message congratulating them and the achievement that they have just accomplished.

## Stopped allowing starting empty workouts
Fitness Duke will now print out an error message if the user attempts to start a workout session even though no workout session has been generated.

### CODE REVIEW

File: src/main/java/seedu/duke/achievements/Achievement.java [hunk 1]
```diff
@@ -1,15 +1,22 @@
 package seedu.duke.achievements;
 
-public class Achievement {
+
+public abstract class Achievement {
     private static final String BLANK = " ";
     private static final String COLON = ":";
-    private static final String nextLine = "\n";
+    private static final String NEXT_LINE = "\n";
```
Comment (EangJS) [inline src/main/java/seedu/duke/achievements/Achievement.java:7]: Taking magic strings to literally 😲

File: src/main/java/seedu/duke/achievements/AchievementListHandler.java [hunk 2]
```diff
@@ -2,42 +2,53 @@
...
+            pushDataToAchievementList(achievementList, achievementTextFile);
         } catch (DukeError e) {
             System.out.println(e.getMessage());
         }
     }
 
+
     private void pushDataToAchievementList(ArrayList<Achievement> achievementList,
                                            File achievementListData) throws DukeError {
         try {
```
Comment (EangJS) [inline src/main/java/seedu/duke/achievements/AchievementListHandler.java:54]: Would be better if you could extend the Storage.java API

File: src/main/java/seedu/duke/achievements/AchievementListHandler.java [hunk 3]
```diff
@@ -2,42 +2,53 @@
...
-        File achievementtextfile = new File(ACHIEVEMENT_LIST_FILE_LOCATION);
-        super.checkForListData(achievementtextfile);
+    public void loadAchievementsFromFile() {
+        File achievementTextFile = new File(ACHIEVEMENT_LIST_FILE_LOCATION);
+        super.checkForListData(achievementTextFile);
         try {
-            pushDataToAchievementList(achievementList, achievementtextfile);
+            pushDataToAchievementList(achievementList, achievementTextFile);
         } catch (DukeError e) {
             System.out.println(e.getMessage());
```
Comment (EangJS) [inline src/main/java/seedu/duke/achievements/AchievementListHandler.java:47]: Try to use logMaster for this.

File: src/main/java/seedu/duke/achievements/AchievementListHandler.java [hunk 4]
```diff
@@ -85,21 +147,81 @@ private Achievement parseAchivement(String achievementInput) throws DukeError {
...
 
+    public ArrayList<Achievement> getAchievementList() {
+        return achievementList;
     }
 
+    public void saveAchievementList() {
+        try {
+            clearFile();
+        } catch (IOException e) {
+            System.out.println(ErrorMessages.ERROR_DELETE_CONTENT_ERROR_TEXT);
```
Comment (EangJS) [inline src/main/java/seedu/duke/achievements/AchievementListHandler.java:200]: Use logger

**Review by EangJS — APPROVED** [review]
Good addition of new features.
**Just asking**, could achievements be calculated during runtime, i.e during start of program based on history of workouts to reduce points of failure?

File: src/main/java/seedu/duke/achievements/AchievementListHandler.java (hunk 3 above)
Comment (ChubbsBunns) [reply src/main/java/seedu/duke/achievements/AchievementListHandler.java:47]: Will do

**Pull request merged**